### PR TITLE
chore: linter fixes for core

### PR DESCRIPTION
**Pull request by Walther**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes missing linting for the `core/test` directory. This meant that e.g. `.only` directives were not caught, despite our intention to fail the linting on the presence of exclusive tests.

**Special notes for your reviewer**:

Previously, in `core`, we were only linting the `src/` directory. Now we will lint the `test/` directory too.

### CODE REVIEW

File: core/package.json [hunk 1]
```diff
@@ -247,7 +247,7 @@
     "clean": "shx rm -rf build",
     "dev": "gulp pegjs-watch",
     "fix-format": "prettier --write \"{src,test}/**/*.ts\"",
-    "lint": "eslint -c ../.eslintrc --ignore-pattern 'src/lib/**' --ext .ts src/",
+    "lint": "eslint -c ../.eslintrc --ignore-pattern 'src/lib/**' --ext .ts src/ test/",
```
Comment (Walther) [inline core/package.json:250]: Previously, in `core`, we were only linting the `src/` directory. Now we will lint the `test/` directory too.

**Review by vvagaytsev — APPROVED** [review]
Nice! 👍 Thank you!

**Review by vvagaytsev — APPROVED** [review]
Nice! 👍 Thank you!

File: core/package.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (vvagaytsev) [reply core/package.json:250]: Or maybe just add `../plugins` as an argument here?

File: core/test/unit/src/cache.ts [hunk 2]
```diff
@@ -252,12 +252,12 @@ describe("BoundedCache", () => {
     // We expect the first 5 keys to have been removed.
     expect(cache["keys"]).to.eql(["5", "6", "7", "8", "9", "10"])
     expect(cache["cache"]).to.eql({
-      "5": "val#5",
-      "6": "val#6",
-      "7": "val#7",
-      "8": "val#8",
-      "9": "val#9",
-      "10": "val#10",
+      5: "val#5",
```
[the remark this answers is not in the document]
Comment (Walther) [reply core/test/unit/src/cache.ts:255]: javascript arrays [are funky](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array#array_indices)

```
/Users/walther/git/garden/core/test/unit/src/cache.ts
  255:7  error  Properties shouldn't be quoted as all quotes are redundant  quote-props
  256:7  error  Properties shouldn't be quoted as all quotes are redundant  quote-props
  257:7  error  Properties shouldn't be quoted as all quotes are redundant  quote-props
  258:7  error  Properties shouldn't be quoted as all quotes are redundant  quote-props
  259:7  error  Properties shouldn't be quoted as all quotes are redundant  quote-props
  260:7  error  Properties shouldn't be quoted as all quotes are redundant  quote-props
  ```

File: core/package.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (Walther) [reply core/package.json:250]: Looks like setting up linting for `plugins` with a simple `../plugins` throws more errors and requires further configuration. I'll add a followup task, let's get this one merged

File: core/test/unit/src/cache.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (thsig) [reply core/test/unit/src/cache.ts:255]: Ugh, haha. I should have used different string keys in these test cases.

**Pull request merged**